beagle_etl/views.py

```diff
@@ -77,6 +77,49 @@ def fix_query_list(self, request_query, key_list):
                     query_dict[single_param] = query_value
         return query_dict
 
+class AssayViewSet(GenericAPIView):
+    serializer_class = AssaySerializer
+    queryset = Assay.objects.all()
+    pagination_class = None
+
+    @swagger_auto_schema(responses={200: AssayElementSerializer})
+    def get(self, request):
+        assay_list = Assay.objects.all()
+        if assay_list and len(assay_list) == 1:
+            assay_response = AssaySerializer(assay_list[0])
+            return Response(assay_response.data, status=status.HTTP_200_OK)
+        error_message = "Assay list has unexpected length of {}".format(len(assay_list))
+        return Response(error_message,status=status.HTTP_404_NOT_FOUND)
+
+    @swagger_auto_schema(request_body=AssayUpdateSerializer,responses={200: AssayElementSerializer})
+    def post(self, request):
+        all_list = request.data.getlist('all')
+        disabled_list = request.data.getlist('disabled')
+        hold_list = request.data.getlist('hold')
+        assay_list = Assay.objects.all()
+        if assay_list and len(assay_list) == 1:
+            assay = assay_list[0]
+            if all_list:
+                assay.all = list(set(all_list))
+            if disabled_list:
+                assay.disabled = list(set(disabled_list))
+            if hold_list:
+                assay.hold = list(set(hold_list))
+            for single_assay in assay.hold:
+                if single_assay in assay.disabled:
+                    error_message = "Assay {} is in both disabled and hold".format(single_assay)
```

Comment:
Only thing which would be really good is going through all assays and join them so error message have list of all assays which are in disabled and hold and also if they missing in the list of all assays. This would enable us to fix all errors if there are more then one.

Reply:
Yeah, that makes sense. I will make it return a list of errors.